### PR TITLE
Exclude test recording files.

### PR DESCRIPTION
**Pull request by sima-zhu**
# Description
Generate job matrix execute before all other jobs, so we it is ideal to be as fast as possible. Scan json files only and exclude the test recording files to speed up the process.
Testing pipeline:
https://github.com/Azure/azure-sdk-for-java/commit/4a11285f65995982b79ecc77bc5f9401f8ec95b9

Resolve https://github.com/Azure/azure-sdk-tools/issues/3126

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/spring/compatibility-tests.yml [hunk 1]
```diff
@@ -10,6 +10,10 @@ stages:
       - template: /eng/common/pipelines/templates/jobs/archetype-sdk-tests-generate.yml
         parameters:
           SparseCheckout: true
+          SparseCheckoutPaths:
```
Comment (benbp) [inline sdk/spring/compatibility-tests.yml:13]: You'll also need to add the python file referenced in the PreGenerationSteps below.

Comment (benbp) [reply sdk/spring/compatibility-tests.yml:13]: Maybe just including `sdk/spring/scripts` is most flexible.

**Review by benbp — APPROVED** [review]

File: eng/pipelines/templates/stages/archetype-sdk-tests.yml [hunk 2]
```diff
@@ -117,6 +117,10 @@ stages:
         jobs:
         - template: /eng/common/pipelines/templates/jobs/archetype-sdk-tests-generate.yml
           parameters:
+            SparseCheckoutPaths:
```
Comment (sima-zhu) [inline eng/pipelines/templates/stages/archetype-sdk-tests.yml:120]: No null checking on `${{ parameters.ServiceDirectory }}`
The script will fail when no service directory passed in. @benbp is this valid case?

Comment (sima-zhu) [reply eng/pipelines/templates/stages/archetype-sdk-tests.yml:120]: Base on my testing, if no service directory pass in, then it does not fail at the this step, but failed at next step as we do not actually checkout the useful files.
The value we pass is `sdk/not-specified/**/*.json`.
IMO, we have to do if/else checking.

Comment (benbp) [reply eng/pipelines/templates/stages/archetype-sdk-tests.yml:120]: What are the cases where service directory is empty?

Comment (sima-zhu) [reply eng/pipelines/templates/stages/archetype-sdk-tests.yml:120]: I don't have any cases. Do we want to remove the if/else check?

**Review by benbp — APPROVED** [review]

File: eng/pipelines/templates/stages/archetype-sdk-tests.yml (hunk 2 above)
Comment (JimSuplizio) [reply eng/pipelines/templates/stages/archetype-sdk-tests.yml:120]: @benbp and @sima-zhu I don't believe this change is correct place to make this change. This change should be made [here](https://github.com/Azure/azure-sdk-for-java/blob/main/eng/common/pipelines/templates/jobs/archetype-sdk-tests-generate.yml#L63) and applied only when the servicedirectory is passed in. The parameter you're overriding here isn't correct. The SparseCheckoutPaths parameter is something that's supposed to be passed through, not something that's supposed to be set here.

Comment (sima-zhu) [reply eng/pipelines/templates/stages/archetype-sdk-tests.yml:120]: Talked offline. This is the place to address this in short run. In long run, recording files move to other repo, then the exclusion is not needed

**Pull request merged**